### PR TITLE
payment-recipe-voucherDefault REM-321

### PR DESCRIPTION
**Pull request by Alinadimmokdad**
rv-dtd    /   pv-dtd

### CODE REVIEW

**Comment by Ghinwa-Arab** [comment]
image
 cash lookup not lighting on validation

**Comment by Ghinwa-Arab** [comment]
image

in rv doc type defaults cash account lookup not lighting on validation

File: src/pages/memos-dtd/[functionId]/form/MemosDtdForm.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
+import { FormControlLabel, Grid, Checkbox } from '@mui/material'
```
Comment (Ghinwa-Arab) [inline src/pages/memos-dtd/[functionId]/form/MemosDtdForm.js:1]: no need to import FormControlLabel,Checkbox,useState

File: src/pages/memos-dtd/[functionId]/form/MemosDtdForm.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      recordId: dtId
+    },
+    maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      dtId: yup.string().required(' ')
+    }),
+    onSubmit: async obj => {
+      const response = await postRequest({
```
Comment (Ghinwa-Arab) [inline src/pages/memos-dtd/[functionId]/form/MemosDtdForm.js:38]: response not used anywhere remove it and call API directly
await postRequest({
        extension: FinancialRepository.FIDocTypeDefaults.set,
        record: JSON.stringify(obj)
      })

File: src/pages/memos-dtd/[functionId]/form/MemosDtdForm.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        extension: FinancialRepository.FIDocTypeDefaults.set,
+        record: JSON.stringify(obj)
+      })
+
+      formik.setValues({
+        ...obj,
+        recordId: obj.dtId
+      })
+
+      toast.success('Record Success')
```
Comment (Ghinwa-Arab) [inline src/pages/memos-dtd/[functionId]/form/MemosDtdForm.js:48]: use plateformLabels

File: src/pages/memos-dtd/[functionId]/index.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    openForm(obj?.dtId)
+  }
+
+  const del = async obj => {
+    await postRequest({
+      extension: FinancialRepository.FIDocTypeDefaults.del,
+      record: JSON.stringify(obj)
+    })
+    invalidate()
+    toast.success('Record Deleted Successfully')
```
Comment (Ghinwa-Arab) [inline src/pages/memos-dtd/[functionId]/index.js:84]: no hard coded toasts, start using plateformLabels

File: src/pages/memos-dtd/[functionId]/index.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        dtId: dtId,
+        maxAccess: access
+      },
+      width: 500,
+      height: 360,
+      title: _labels.doctypeDefault
+    })
+  }
+
+  console.log(functionId, 'func index')
```
Comment (Ghinwa-Arab) [inline src/pages/memos-dtd/[functionId]/index.js:102]: remove

File: src/pages/pv-dtd/form/PVDocTypeDefaultsForm.js [hunk 6]
```diff
@@ -0,0 +1,179 @@
+import { FormControlLabel, Grid, Checkbox } from '@mui/material'
+import { useContext, useEffect, useState } from 'react'
```
Comment (Ghinwa-Arab) [inline src/pages/pv-dtd/form/PVDocTypeDefaultsForm.js:2]: no need to import useState

File: src/pages/pv-dtd/form/PVDocTypeDefaultsForm.js [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+      disableExpenseGrid: false
+    },
+    maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      dtId: yup.string().required(' ')
+    }),
+    onSubmit: async obj => {
+      const response = await postRequest({
```
Comment (Ghinwa-Arab) [inline src/pages/pv-dtd/form/PVDocTypeDefaultsForm.js:44]: remove const response, call api directly

File: src/pages/pv-dtd/form/PVDocTypeDefaultsForm.js [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+        extension: FinancialRepository.FIDocTypeDefaults.set,
+        record: JSON.stringify(obj)
+      })
+
+      formik.setValues({
+        ...obj,
+        recordId: obj.dtId
+      })
+
+      toast.success('Record Success')
```
Comment (Ghinwa-Arab) [inline src/pages/pv-dtd/form/PVDocTypeDefaultsForm.js:54]: use plateform labels

File: src/pages/pv-dtd/index.js [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+    openForm(obj?.dtId)
+  }
+
+  const del = async obj => {
+    await postRequest({
+      extension: FinancialRepository.FIDocTypeDefaults.del,
+      record: JSON.stringify(obj)
+    })
+    invalidate()
+    toast.success('Record Deleted Successfully')
```
Comment (Ghinwa-Arab) [inline src/pages/pv-dtd/index.js:83]: use plateformLabels

File: src/pages/pv-dtd/index.js [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+  }
+
+  function openForm(dtId) {
+    stack({
+      Component: PVDocTypeDefaultsForm,
+      props: {
+        labels: _labels,
+        dtId: dtId,
```
Comment (Ghinwa-Arab) [inline src/pages/pv-dtd/index.js:91]: enough to pass dtId,

File: src/pages/rv-dtd/form/RVDocTypeDefaultsForm.js [hunk 11]
```diff
@@ -0,0 +1,144 @@
+import { Grid } from '@mui/material'
+import { useContext, useEffect, useState } from 'react'
```
Comment (Ghinwa-Arab) [inline src/pages/rv-dtd/form/RVDocTypeDefaultsForm.js:2]: no need to import useState

File: src/pages/rv-dtd/form/RVDocTypeDefaultsForm.js [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+    maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      dtId: yup.string().required(' '),
+
+      cashAccountId: yup.string().required(' ')
+    }),
+    onSubmit: async obj => {
+      const response = await postRequest({
```
Comment (Ghinwa-Arab) [inline src/pages/rv-dtd/form/RVDocTypeDefaultsForm.js:38]: remove const response

File: src/pages/rv-dtd/form/RVDocTypeDefaultsForm.js [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+        extension: FinancialRepository.FIDocTypeDefaults.set,
+        record: JSON.stringify(obj)
+      })
+
+      formik.setValues({
+        ...obj,
+        recordId: obj.dtId
+      })
+
+      toast.success('Record Success')
```
Comment (Ghinwa-Arab) [inline src/pages/rv-dtd/form/RVDocTypeDefaultsForm.js:48]: platformLabels

File: src/pages/rv-dtd/index.js [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+    openForm(obj?.dtId)
+  }
+
+  const del = async obj => {
+    await postRequest({
+      extension: FinancialRepository.FIDocTypeDefaults.del,
+      record: JSON.stringify(obj)
+    })
+    invalidate()
+    toast.success('Record Deleted Successfully')
```
Comment (Ghinwa-Arab) [inline src/pages/rv-dtd/index.js:81]: plateformLabels

File: src/pages/rv-dtd/index.js [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+  }
+
+  function openForm(dtId) {
+    stack({
+      Component: RVDocTypeDefaultsForm,
+      props: {
+        labels: _labels,
+        dtId: dtId,
```
Comment (Ghinwa-Arab) [inline src/pages/rv-dtd/index.js:89]: dtId, is enough

File: src/pages/memos-dtd/[functionId]/form/MemosDtdForm.js [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+    validationSchema: yup.object({
+      dtId: yup.string().required(' ')
+    }),
+    onSubmit: async obj => {
+      await postRequest({
+        extension: FinancialRepository.FIDocTypeDefaults.set,
+        record: JSON.stringify(obj)
+      })
+
+      formik.setValues({
```
Comment (hadichahine) [inline src/pages/memos-dtd/[functionId]/form/MemosDtdForm.js:45]: No need sir. You already have the dtId inside values.

File: src/pages/pv-dtd/form/PVDocTypeDefaultsForm.js [hunk 17]
```diff
@@ -0,0 +1,181 @@
...
+    validationSchema: yup.object({
+      dtId: yup.string().required(' ')
+    }),
+    onSubmit: async obj => {
+      await postRequest({
+        extension: FinancialRepository.FIDocTypeDefaults.set,
+        record: JSON.stringify(obj)
+      })
+
+      formik.setValues({
```
Comment (hadichahine) [inline src/pages/pv-dtd/form/PVDocTypeDefaultsForm.js:51]: We don't need this.

**Review by hadichahine — CHANGES_REQUESTED** [review]

File: src/pages/rv-dtd/form/RVDocTypeDefaultsForm.js [hunk 18]
```diff
@@ -0,0 +1,144 @@
...
+
+      cashAccountId: yup.string().required(' ')
+    }),
+    onSubmit: async obj => {
+      await postRequest({
+        extension: FinancialRepository.FIDocTypeDefaults.set,
+        record: JSON.stringify(obj)
+      })
+
+      formik.setValues({
```
[the remark this answers is not in the document]
Comment (Alinadimmokdad) [reply src/pages/rv-dtd/form/RVDocTypeDefaultsForm.js:45]: just to give value for recirdId

**Review by hadichahine — APPROVED** [review]

**Pull request merged**